### PR TITLE
Implement truck class

### PR DESCRIPTION
**Pull request by Abdiiir**
- Color dependent on car type ( no scaling of color )

### CODE REVIEW

File: src/sim.py [hunk 1]
```diff
@@ -57,6 +58,8 @@ def step(self):
                 elif type(car) == AutonomousCar:
                     autonomous_car_env = self.highway.get_autonomous_car_env(car_ind, lane_ind)
                     car.take_action(self.step_time, car_env, autonomous_car_env)
+                elif type(car) == Truck:
```
Comment (mactat) [inline src/sim.py:61]: isn't it the same function in both classes?

Comment (mactat) [reply src/sim.py:61]: or is there a difference between `car.take_action(self.step_time)` in case of normal car and truck

File: src/classes.py [hunk 2]
```diff
@@ -166,10 +167,11 @@ def switch_lane(self, direction):
 this car.
 '''
 class AutonomousCar(Car):
-    def __init__(self, initial_speed, lane, drivers_mood=0.95, number=0, acc=0, breaking=0,radius = 1000, delay = 0):
```
Comment (mactat) [inline src/classes.py:169]: What is the reason behind this change?

**Review by mactat — CHANGES_REQUESTED** [review]

File: src/sim.py (hunk 1 above)
Comment (Abdiiir) [reply src/sim.py:61]: No, a boolean is added to ` choose_speed` fun. When it is a Truck to limit the speed.

File: src/classes.py (hunk 2 above)
Comment (Abdiiir) [reply src/classes.py:169]: This should not be there (y)

File: src/sim.py [hunk 3]
```diff
@@ -52,7 +53,7 @@ def step(self):
                 car_env = self.highway.get_car_env(car_ind, lane_ind)
 
                 # Make changes in car, as speed, changing lane, etc based on env
-                if type(car) == Car:
+                if type(car) == (Car or Truck):
```
Comment (mactat) [inline src/sim.py:56]: `(Car or Truck)` is always false. Use `type(car) == Car or type(car) == Truck`

File: src/sim.py [hunk 4]
```diff
@@ -115,14 +119,20 @@ def get_random_vehicle(self, lane):
                                     drivers_mood = 0,
                                     radius = 1000, 
                                     delay = 0)
+        elif vehicle_type == Truck:
+            return Truck(self.choose_speed(True),
+                                    lane = lane,
+                                    number = self.in_car_counter,
+                                    drivers_mood = 0.2, # Not sure?
```
Comment (mactat) [inline src/sim.py:126]: more like 0.95. Higher value means driver is less likely to take an action

File: src/classes.py [hunk 5]
```diff
@@ -190,6 +192,15 @@ def take_action(self, time_elapsed, car_env, autonomous_env=None):
...
 
+'''
+Truck class that extends Car.
+Truck itself do not make any decision, it has to ask the driver. (Similar to Car class)
+'''
+class Truck(Car):
+    def __init__(self, initial_speed, lane, drivers_mood=0.95, number=0, acc=0, breaking=0):
+        super().__init__(initial_speed, lane, drivers_mood, number, acc, breaking)
+        self.color = 3 # Red 
+
```
Comment (mactat) [inline src/classes.py:203]: We have to limit max speed of the truck to 90km/h or 85km/h, so i would modify the driver decide function to change the always speed_limit parameter to 90 and call the same function from parent class

**Review by mactat — CHANGES_REQUESTED** [review]

**Comment by mactat** [comment]
`vehicle_type = np.random.choice([Car,AutonomousCar, Truck], 1, p=[1-(self.propotion_of_trucks + self.propotion_of_autonomous),self.propotion_of_trucks,self.propotion_of_autonomous])` <- wrong order here

**Review by mohammadnabilo — APPROVED** [review]
LGTB+

**Review by mactat — APPROVED** [review]

**Comment by mactat** [comment]
l_2_a_30_m_85
